cmd/jiralert/main.go

```diff
@@ -94,16 +97,27 @@ func main() {
 		level.Debug(logger).Log("msg", "  matched receiver", "receiver", conf.Name)
 
 		// TODO: Consider reusing notifiers or just jira clients to reuse connections.
-		tp := jira.BasicAuthTransport{
-			Username: conf.User,
-			Password: string(conf.Password),
+
+		// if cert and key are specified, ignore username/password
+		// TODO can we (easily) support both client cert and username/password?
+		hc, err := httpClient(conf)
+		if err != nil {
+			errorHandler(w, http.StatusInternalServerError, err, conf.Name, &data, logger)
+			return
 		}
-		client, err := jira.NewClient(tp.Client(), conf.APIURL)
+
+		client, err := jira.NewClient(hc, conf.APIURL)
 		if err != nil {
 			errorHandler(w, http.StatusInternalServerError, err, conf.Name, &data, logger)
 			return
 		}
 
+		if conf.User != "" && conf.Password != "" {
+			// SetBasicAuth is marked as deprecated, but can't use BasicAuthTransport
```

Comment:
This is ok for now, just let's mark some ignore (there must be some ignore statement in comment) for staticcheck for now. 

I would be surprise though that there is no way of doing this :thinking:

Reply:
BTW, do you need basic auth with custom TLS (e.g with client cert)?

Reply:
I went back and forth re: mTLS + Basic Auth - in my setup only the client cert will be used but in theory both could be used (e.g., a generic cert for access plus a specific username/password?  I'm hand-waving...).  In the end, I believe Prometheus itself allows both for scrape configs, so I went with supporting both.

Reply:
The incantation is:

//lint:ignore <staticcheck code> <required comment>

(no spaces allowed between // and lint:ignore)